.github/workflows/on_pr_opened.yaml

```diff
@@ -19,6 +19,7 @@ concurrency:
 jobs:
   build:
     name: Build & Push
+    if: "github.event.pull_request.author_association != 'COLLABORATOR' && github.event.review.author_association != 'OWNER'"
```

Comment:
this prevents the non collaborator workflow from running for a collaborator. the collaborator pull requests are automatically deployed in a separate workflow.